google/cloud/spanner_dbapi/cursor.py

```diff
@@ -309,19 +314,28 @@ def executemany(self, operation, seq_of_params):
         self._result_set = None
         self._row_count = _UNSET_COUNT
 
-        class_ = parse_utils.classify_stmt(operation)
-        if class_ == parse_utils.STMT_DDL:
+        parsed_statement = parse_utils._classify_stmt(operation)
+        if parsed_statement.statement_type == StatementType.DDL:
             raise ProgrammingError(
                 "Executing DDL statements with executemany() method is not allowed."
             )
 
         # For every operation, we've got to ensure that any prior DDL
         # statements were run.
         self.connection.run_prior_DDL_statements()
+        if parsed_statement.statement_type == StatementType.CLIENT_SIDE:
```

Comment:
I think this code block should be moved to before the `run_prior_DDL_statements`. Now it is (theoretically?) possible that the DDL statements are executed, and then we return an error.

Reply:
Done